### PR TITLE
Build process and CI improvements

### PR DESCRIPTION
**Pull request by invidian**
See commit messages for details.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.PHONY:	all release-bin image clean test vendor ci lint-bin codespell
+.PHONY:	all image clean test vendor ci lint-bin codespell
```
Comment (knrt10) [inline Makefile]:
```suggestion
.PHONY: all image clean test vendor ci lint-bin codespell
```

File: .golangci.yml [hunk 2]
```diff
@@ -11,6 +11,10 @@ issues:
     - linters:
         - gomnd
       source: "klog.V\\("
```
Comment (knrt10) [inline .golangci.yml:13]: Nit commit message
```patch
-contianed
+contained
```

**Review by knrt10 — CHANGES_REQUESTED** [review]
Small NIT, rest LGTM

File: .golangci.yml (hunk 2 above)
Comment (invidian) [reply .golangci.yml:13]: Thanks, fixed.

File: Makefile (hunk 1 above)
Comment (invidian) [reply Makefile]: Well, it was originally like this, but I fixed it in separate commit.

**Review by knrt10 — APPROVED** [review]

**Pull request merged**